### PR TITLE
Bump k8s to v1.27.1

### PR DESCRIPTION
**Pull request by kashifest**
Apart from bumping to k8s to v1.27.1 and starting upgrade from v1.26.4, this PR also bumps renovate to allow newer minor version related PRs.

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -56,13 +56,13 @@
       "matchPackageNames": [
         "kindest/node"
       ],
-      "allowedVersions": "<=1.26"
+      "allowedVersions": "<=1.27"
```
Comment (furkatgofurov7) [inline renovate.json:59]: not sure if these are ever used (usually bump was done manually)

**Review by furkatgofurov7 — APPROVED** [review]
/approve

/hold in case some more reviews are neeeded

**Comment by furkatgofurov7** [comment]
**Note:** CAPI support for k8s 1.27 is available in recent [v1.4.2](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.4.2) release and CI is already using it (main/release-1.4 jobs), but it would be great to bump it also in CAPM3/IPAM [go mod](https://github.com/metal3-io/cluster-api-provider-metal3/blob/release-1.4/go.mod#L23)

File: renovate.json (hunk 1 above)
Comment (kashifest) [reply renovate.json:59]: true, we do it manually, I was also thinking about removing them but kept is just as a reminder.

**Comment by kashifest** [comment]
> **Note:** CAPI support for k8s 1.27 is available in recent [v1.4.2](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.4.2) release and CI is already using it (main/release-1.4 jobs), but it would be great to bump it also in CAPM3/IPAM [go mod](https://github.com/metal3-io/cluster-api-provider-metal3/blob/release-1.4/go.mod#L23)

Thanks I have PRs now to bump CAPI to v1.4.2 in CAPM3/IPAM

**Comment by furkatgofurov7** [comment]
> > **Note:** CAPI support for k8s 1.27 is available in recent [v1.4.2](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.4.2) release and CI is already using it (main/release-1.4 jobs), but it would be great to bump it also in CAPM3/IPAM [go mod](https://github.com/metal3-io/cluster-api-provider-metal3/blob/release-1.4/go.mod#L23)
> 
> Thanks I have PRs now to bump CAPI to v1.4.2 in CAPM3/IPAM

Thanks, and feel free to unhold :)

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Pull request merged**